connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntitySetPassengersTranslator.java

```diff
@@ -99,66 +101,149 @@ public void translate(ServerEntitySetPassengersPacket packet, GeyserSession sess
+        return mountedHeightOffset;
+    }
+
+    private float getHeightOffset(Entity passenger) {
+        boolean isBaby;
+        switch (passenger.getEntityType()) {
+            case SKELETON:
+            case STRAY:
+            case WITHER_SKELETON:
+                return -0.6f;
+            case ARMOR_STAND:
+                // Armor stand isn't a marker
+                if (passenger.getMetadata().getFloat(EntityData.BOUNDING_BOX_HEIGHT) != 0.0f) {
+                    return 0.1f;
+                } else {
+                    return 0.0f;
+                }
+            case ENDERMITE:
+            case SILVERFISH:
+                return 0.1f;
+            case PIGLIN:
+            case PIGLIN_BRUTE:
+            case ZOMBIFIED_PIGLIN:
+                isBaby = passenger.getMetadata().getFlags().getFlag(EntityFlag.BABY);
+                return isBaby ? -0.05f : -0.45f;
+            case ZOMBIE:
+                isBaby = passenger.getMetadata().getFlags().getFlag(EntityFlag.BABY);
+                return isBaby ? 0.0f : -0.45f;
+            case EVOKER:
+            case ILLUSIONER:
+            case PILLAGER:
+            case RAVAGER:
+            case VINDICATOR:
+            case WITCH:
+                return -0.45f;
+            case PLAYER:
+                return -0.35f;
         }
+        return 0f;
+    }
+
+    private void updateOffset(Entity passenger, Entity mount, GeyserSession session, boolean rider, boolean riding, boolean moreThanOneEntity) {
         passenger.getMetadata().getFlags().setFlag(EntityFlag.RIDING, riding);
         if (riding) {
+            // Without the Y offset, Bedrock players will find themselves in the floor when mounting
+            float mountedHeightOffset = getMountedHeightOffset(mount);
+            float heightOffset = getHeightOffset(passenger);
+
+            float xOffset = 0;
+            float yOffset = mountedHeightOffset + heightOffset;
+            float zOffset = 0;
+            switch (mount.getEntityType()) {
+                case BOAT:
+                    // Without the X offset, more than one entity on a boat is stacked on top of each other
+                    if (rider && moreThanOneEntity) {
+                        xOffset = 0.2f;
+                    } else if (moreThanOneEntity) {
+                        xOffset = -0.6f;
+                    }
+                    break;
+                case CHICKEN:
+                    zOffset = -0.1f;
+                    break;
+                case LLAMA:
+                    zOffset = -0.3f;
+                    break;
+            }
+            /*
+             * Bedrock Differences
+             * Zoglin & Hoglin seem to be taller in Bedrock edition
+             * Horses are tinier
+             * Players, Minecarts, and Boats have different origins
+             */
+            if (passenger.getEntityType() == EntityType.PLAYER) {
+                yOffset += 1.6f;
```

Comment:
If this is specifically 1.62, maybe set this to `EntityType.PLAYER.getOffset()`?